### PR TITLE
Add functions pretrain_model and get_transformer_encoder.

### PR DESCRIPTION
**Pull request by boru-roylu**

### CODE REVIEW

File: src/mmt_models.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  )
+  return mmt_models.MmtEncoder(**kwargs)
+
+
+def pretrain_model(mmt_config: configs.MmtConfig,
+                   mpp_output_num_classes: int,
+                   patch_embedding_size: int,
+                   return_core_pretrainer_model: bool = False):
+  """Returns model to be used for pre-training.
+  Args:
```
Comment (wqfcr-google) [inline src/mmt_models.py:60]: Nit: missing patch_embedding_size

**Review by wqfcr-google — APPROVED** [review]
LGTM.

File: src/mmt_models.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      dtype=tf.int32)
+  masked_patch_label_weights = tf.keras.layers.Input(
+      shape=(None,),
+      name='masked_patch_label_weights',
+      dtype=tf.int32)
+
+  transformer_encoder = get_transformer_encoder(mmt_config)
+  transformer_encoder._word_embedding_layer.build(word_ids.shape)
+
+  pretrainer_model = mmt_models.MmtPretrainer(
```
Comment (santiaontanon-google) [inline src/mmt_models.py:120]: nit: is "pretrainer" the right name here? I see "pretraining_model" more often. If you prefer "pretrainer", it's fine, but it sounds weird.

**Review by santiaontanon-google — APPROVED** [review]
Thanks Roy, looks good! One general comment is to fill the pull request description with a description of the changes, in addition to the title of the pull request. Which can help us review the changes.

File: src/mmt_models.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+import tensorflow as tf
+from official.modeling import tf_utils
+
+import configs
+import layers as mmt_layers
+import models as mmt_models
+
+
+
```
Comment (jtainslie) [inline src/mmt_models.py]: Minor nit: 2 blank lines is more standard than 3.

File: src/mmt_models.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+def get_transformer_encoder(mmt_config):
+  """Gets a `MmtEncoder` object.
+  Args:
```
Comment (jtainslie) [inline src/mmt_models.py:26]: Nit: there's usually a blank line between the first line and `Args:`, and also before the `Returns:` section.  (Same for docstrings below too.)  https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

File: src/mmt_models.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+import tensorflow as tf
+from official.modeling import tf_utils
+
+import configs
+import layers as mmt_layers
+import models as mmt_models
+
+
+
+def get_transformer_encoder(mmt_config):
```
Comment (jtainslie) [inline src/mmt_models.py]: Nit:
```suggestion
def get_transformer_encoder(mmt_config: configs.MmtConfig):
```

**Review by jtainslie — APPROVED** [review]

**Pull request merged**